pkg/controller/controller_utils.go

```diff
@@ -430,14 +438,35 @@ func GetPodFromTemplate(template *api.PodTemplateSpec, parentObject runtime.Obje
 			GenerateName: prefix,
 		},
 	}
+	if setControllerRef {
+		typeAccessor, err := meta.TypeAccessor(parentObject)
+		if err != nil {
+			return nil, fmt.Errorf("parentObject does not have TypeMeta, %v", err)
+		}
+		if len(typeAccessor.GetAPIVersion()) == 0 {
```

Comment:
nit: checking len makes me think GetApiVersion returns an array, any reason to not simply check empty string? also can it every not have a kind and apiversion?